### PR TITLE
Feature/fix transfer counts

### PR DESCRIPTION
**Pull request by karthikeyan-cere**

### CODE REVIEW

File: backend/docker/sql/backend.sql [hunk 1]
```diff
@@ -333,4 +333,35 @@ CREATE TRIGGER event_count_trunc AFTER TRUNCATE ON event
...
-COMMIT;
\ No newline at end of file
+COMMIT;
+
+-- Transfers
+START TRANSACTION;
+CREATE FUNCTION transfer_count() RETURNS trigger LANGUAGE plpgsql AS
+$$BEGIN
+  IF TG_OP = 'INSERT' THEN
+    IF (NEW.method = 'transfer') OR (NEW.method = 'transferKeepAlive') THEN
```
Comment (AndreiNavoichyk) [inline backend/docker/sql/backend.sql]: Let's use `in` here and everywhere.

File: backend/docker/sql/backend.sql [hunk 2]
```diff
@@ -333,4 +333,35 @@ CREATE TRIGGER event_count_trunc AFTER TRUNCATE ON event
...
+      UPDATE total SET count = count + 1 WHERE name = 'transfers';
+      RETURN NEW;
+    END IF;
```
Comment (AndreiNavoichyk) [inline backend/docker/sql/backend.sql]: Should you use `ELSE` branch here (line 352-355)? I think so. Not to copy-paste, please, use `IF` with a composite condition. Fix here and everywhere.

File: backend/docker/sql/backend.sql [hunk 3]
```diff
@@ -333,4 +333,35 @@ CREATE TRIGGER event_count_trunc AFTER TRUNCATE ON event
...
+CREATE CONSTRAINT TRIGGER transfer_count_mod
+  AFTER INSERT OR DELETE ON extrinsic
+  DEFERRABLE INITIALLY DEFERRED
+  FOR EACH ROW EXECUTE PROCEDURE transfer_count();
+-- TRUNCATE triggers must be FOR EACH STATEMENT
+CREATE TRIGGER transfer_count_trunc AFTER TRUNCATE ON extrinsic
+  FOR EACH STATEMENT EXECUTE PROCEDURE transfer_count();
+-- initialize the counter table
+UPDATE total SET count = (SELECT count(*) FROM extrinsic WHERE method = 'transfer' or method = 'transferKeepAlive') WHERE name = 'transfers';
+COMMIT;
```
Comment (AndreiNavoichyk) [inline backend/docker/sql/backend.sql]: Copy this code to `backend-dev` file as well.

File: backend/docker/sql/backend.sql (hunk 1 above)
Comment (karthikeyan-cere) [reply backend/docker/sql/backend.sql]: Updated

File: backend/docker/sql/backend.sql (hunk 3 above)
Comment (karthikeyan-cere) [reply backend/docker/sql/backend.sql]: added

File: backend/docker/sql/backend.sql (hunk 2 above)
Comment (karthikeyan-cere) [reply backend/docker/sql/backend.sql]: Updated

File: backend/docker/sql/backend-dev.sql [hunk 4]
```diff
@@ -334,3 +334,32 @@ CREATE TRIGGER event_count_trunc AFTER TRUNCATE ON event
...
+    IF TG_OP = 'INSERT' THEN
+      UPDATE total SET count = count + 1 WHERE name = 'transfers';
+      RETURN NEW;
+    ELSIF TG_OP = 'DELETE' THEN
+      UPDATE total SET count = count - 1 WHERE name = 'transfers';
+      RETURN OLD;
+    ELSE
+      UPDATE total SET count = 0 WHERE name = 'transfers';
+      RETURN NULL;
+    END IF;
```
Comment (AndreiNavoichyk) [inline backend/docker/sql/backend-dev.sql:352]: We can move this logic to line 354 and write it once.

Comment (karthikeyan-cere) [reply backend/docker/sql/backend-dev.sql:352]: We cant move the else block to line 345, if we move then other than method `transfer` and `transferKeepAlive` the transfers count in total will become zero.

**Review by AndreiNavoichyk — APPROVED** [review]

**Pull request merged**